integration_tests/sdk/data_integration_tests/s3_test.py

```diff
@@ -0,0 +1,411 @@
+from typing import Optional
+
+import pandas as pd
+import pytest
+from aqueduct.artifacts.base_artifact import BaseArtifact
+from aqueduct.constants.enums import ArtifactType
+from aqueduct.error import AqueductError, InvalidUserArgumentException
+from aqueduct.integrations.s3_integration import S3Integration
+
+from sdk.data_integration_tests.flow_manager import FlowManager
+from sdk.data_integration_tests.s3_data_validator import S3DataValidator
+from sdk.data_integration_tests.save import save
+from sdk.data_integration_tests.validation_helpers import (
+    check_hotel_reviews_table_artifact,
+    check_hotel_reviews_table_data,
+)
+from sdk.shared.globals import artifact_id_to_saved_identifier
+from sdk.shared.naming import generate_object_name, generate_table_name
+from sdk.shared.validation import check_artifact_was_computed
+
+
+@pytest.fixture(autouse=True)
+def assert_data_integration_is_s3(data_integration):
+    assert isinstance(data_integration, S3Integration)
+
+
+def _save_artifact_and_check(
+    flow_manager: FlowManager,
+    data_integration: S3Integration,
+    artifact: BaseArtifact,
+    format: Optional[str],
+    object_identifier: Optional[str] = None,
+):
+    """Saves the artifact by publishing a flow, and then checks that the data now exists in S3."""
+    assert isinstance(artifact, BaseArtifact)
+
+    if object_identifier is None:
+        object_identifier = generate_table_name() if format is not None else generate_object_name()
+    save(data_integration, artifact, object_identifier, format)
+
+    flow = flow_manager.publish_flow_test(artifact)
+
+    S3DataValidator(flow_manager._client, data_integration).check_saved_artifact_data(
+        flow, artifact.id(), artifact.type(), format, expected_data=artifact.get()
+    )
+
+
+def test_s3_table_fetch_and_save(flow_manager, data_integration):
```

Comment:
I personally prefer to use `s3_integration` or add a typehint (not sure if linter will complain). This way it's more natural that we call the s3 specific `.file` API in the body

Reply:
but not a big deal :)

Reply:
We can't use `s3_integration` because `data_integration` is the name of the pytest fixture (configurable by the `--data` flag), and we want to use that fixture through these test suites.

The `assert_data_integration_is_s3()` fixture at the top of this file will ensure that the data_integration is an S3Integration for all tests in this file. I plan to add type checking to our integration tests in the near future (the task is prioritized in the project).